spinalcordtoolbox/deepseg/inference.py

```diff
@@ -306,6 +306,12 @@ def segment_nnunet(path_img, tmpdir, predictor, device: torch.device):
     if is_rootlet_model:
         targets = ["_rootlets"]
         outputs = [img_out]
+    # for spinal cord segmentation models with only 1 output label, save the image directly with specific "_seg" suffix
+    # this is added to preserve the typical expected output for SC models (`sct_propseg`, `sct_deepseg_sc`, etc.)
+    # see also: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4805
+    elif list(labels.keys()) == ['sc']:
```

Comment:
It's a bit redundant here, because we're comparing to a list of length 1, but I generally like to make these kinds of comparisons deterministic by using `sorted`:
```suggestion
    elif sorted(labels.keys()) == ['sc']:
```

Reply:
Neat! I didn't realize that `sorted` on a `dict_keys` object returns a `list`. Definitely internalizing that, thanks!

```python
>>> labels = {'sc': 1}
>>> labels.keys()
dict_keys(['sc'])
>>> sorted(labels.keys())
['sc']